src/kem/sike/P434/AMD64/fp_x64_asm.S

```diff
@@ -0,0 +1,939 @@
+    mulx   \T7, \T8, 16\M1   // T7:T8 = A1*B2
+    adc    \T6, \T8        
+    mulx   \T8, \T9, 24\M1   // T8:T9 = A1*B3
+    adc    \T7, \T9        
+    adc    \T8, 0         
+
+    add    \T4, \T0
+    mov    8\C, \T4          // C1_final
+    adc    \T5, \T1
+    adc    \T6, \T2
+    adc    \T7, \T3
+    mov    rdx, 16\M0
+    adc    \T8, 0
+
+    mulx   \T1, \T0, \M1     // T1:T0 = A2*B0
+    mulx   \T2, \T3, 8\M1    // T2:T3 = A2*B1
+    add    \T1, \T3        
+    mulx   \T3, \T4, 16\M1   // T3:T4 = A2*B2
+    adc    \T2, \T4        
+    mulx   \T4,\T9, 24\M1    // T3:T4 = A2*B3
+    adc    \T3, \T9        
+    mov    rdx, 24\M0
+    adc    \T4, 0          
+
+    add    \T0, \T5
+    mov    16\C, \T0         // C2_final
+    adc    \T1, \T6
+    adc    \T2, \T7
+    adc    \T3, \T8
+    adc    \T4, 0
+
+    mulx   \T5, \T0, \M1     // T5:T0 = A3*B0
+    mulx   \T6, \T7, 8\M1    // T6:T7 = A3*B1
+    add    \T5, \T7        
+    mulx   \T7, \T8, 16\M1   // T7:T8 = A3*B2
+    adc    \T6, \T8        
+    mulx   \T8, \T9, 24\M1   // T8:T9 = A3*B3
+    adc    \T7, \T9         
+    adc    \T8, 0         
+
+    add    \T1, \T0
+    mov    24\C, \T1         // C3_final
+    adc    \T2, \T5
+    mov    32\C, \T2         // C4_final
+    adc    \T3, \T6
+    mov    40\C, \T3         // C5_final
+    adc    \T4, \T7
+    mov    48\C, \T4         // C6_final
+    adc    \T8, 0
+    mov    56\C, \T8         // C7_final
+.endm
+#endif
+
+
+//*****************************************************************************
+//  434-bit multiplication using Karatsuba (one level), schoolbook (one level)
+//***************************************************************************** 
+.global mul434_asm
+mul434_asm:    
+    push   r12
+    push   r13 
+    push   r14 
+    push   r15
+    mov    rcx, reg_p3 
+
+    // r8-r11 <- AH + AL, rax <- mask
+    xor    rax, rax
+    mov    r8, [reg_p1]
+    mov    r9, [reg_p1+8]
+    mov    r10, [reg_p1+16]
+    mov    r11, [reg_p1+24] 
+    push   rbx 
+    push   rbp
+    sub    rsp, 96
+    add    r8, [reg_p1+32]
+    adc    r9, [reg_p1+40]
+    adc    r10, [reg_p1+48]
+    adc    r11, 0
+    sbb    rax, 0
+    mov    [rsp], r8
+    mov    [rsp+8], r9
+    mov    [rsp+16], r10
+    mov    [rsp+24], r11
+
+    // r12-r15 <- BH + BL, rbx <- mask
+    xor    rbx, rbx
+    mov    r12, [reg_p2]
+    mov    r13, [reg_p2+8]
+    mov    r14, [reg_p2+16]
+    mov    r15, [reg_p2+24]
+    add    r12, [reg_p2+32]
+    adc    r13, [reg_p2+40]
+    adc    r14, [reg_p2+48]
+    adc    r15, 0
+    sbb    rbx, 0
+    mov    [rsp+32], r12
+    mov    [rsp+40], r13
+    mov    [rsp+48], r14
+    mov    [rsp+56], r15
+    
+    // r12-r15 <- masked (BH + BL)
+    and    r12, rax
+    and    r13, rax
+    and    r14, rax
+    and    r15, rax
+
+    // r8-r11 <- masked (AH + AL)
+    and    r8, rbx
+    and    r9, rbx
+    and    r10, rbx
+    and    r11, rbx
+
+    // r8-r11 <- masked (AH + AL) + masked (AH + AL)
```

Comment:
but that's actually ``masked (AH + AL) + masked (BH + BL)``

Reply:
This code is coming from https://github.com/Microsoft/PQCrypto-SIDH so I'd want to check with Patrick Longa to see what should be done here.

Reply:
https://github.com/microsoft/PQCrypto-SIDH/pull/23